pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -43,22 +43,24 @@ type objectWatcherImpl struct {
 	ClusterClientSetFunc ClientSetFunc
 	resourceInterpreter  resourceinterpreter.ResourceInterpreter
 	InformerManager      genericmanager.MultiClusterInformerManager
+	ClientOption         *util.ClientOption
 }
 
 // NewObjectWatcher returns an instance of ObjectWatcher
-func NewObjectWatcher(kubeClientSet client.Client, restMapper meta.RESTMapper, clusterClientSetFunc ClientSetFunc, interpreter resourceinterpreter.ResourceInterpreter) ObjectWatcher {
+func NewObjectWatcher(kubeClientSet client.Client, restMapper meta.RESTMapper, clusterClientSetFunc ClientSetFunc, interpreter resourceinterpreter.ResourceInterpreter, clientOption *util.ClientOption) ObjectWatcher {
```

Comment:
This `clientOption` might be confusing with `kubeClientSet`. I think we can change the signature.